lib/vdc/DCWRF.cpp

```diff
@@ -748,42 +748,38 @@ int DCWRF::_GetProj4String(
 // the files as being "idealized"; for idealized cases the horizontal 
 // coordinates are initialized to constant values.
 //
-bool DCWRF::_isConstantHorizontalCoords(NetCDFCollection *ncdfc) const {
+bool DCWRF::_isConstantValuedVariable(NetCDFCollection *ncdfc, string varname) const {
 
 
 	bool enabled = EnableErrMsg(false);
 
-	vector <string> coordVars = {"XLONG", "XLAT"};
 	vector <float> data;
-	for( const auto& c : coordVars ) {
 
-		vector <size_t> dims = ncdfc->GetSpatialDims(c);
-		VAssert(dims.size() == 2);
+	vector <size_t> dims = ncdfc->GetSpatialDims(varname);
 
-		vector <size_t> start = {0,0};
-		vector <size_t> count = {dims[0],dims[1]};
+	vector <size_t> start(dims.size(), 0);
+	vector <size_t> count = dims;
 
-		data.resize(Wasp::VProduct(dims));
-		
-		int fd = ncdfc->OpenRead(0, c);
-		if (fd<0) {
-			EnableErrMsg(enabled);
-			return(false);
-		}
+	data.resize(Wasp::VProduct(dims));
+	
+	int fd = ncdfc->OpenRead(0, varname);
+	if (fd<0) {
+		EnableErrMsg(enabled);
+		return(false);
+	}
 
-		int rc = ncdfc->Read(start.data(), count.data(), data.data(), fd);
-		if (rc<0) {
-			EnableErrMsg(enabled);
-			return(false);
-		}
+	int rc = ncdfc->Read(start.data(), count.data(), data.data(), fd);
+	if (rc<0) {
+		EnableErrMsg(enabled);
+		return(false);
+	}
 
-		ncdfc->Close(fd);
+	ncdfc->Close(fd);
 
-		for (size_t i=0; i<Wasp::VProduct(dims)-1; i++) {
-			if (data[i] != data[i+1]) {
-				EnableErrMsg(enabled);
-				return(false);
-			}
+	for (size_t i=0; i<Wasp::VProduct(dims)-1; i++) {
+		if (data[i] != data[i+1]) {
+			EnableErrMsg(enabled);
```

Comment:
Do we want to account for "almost constant" since these are floats? Also, performance can be improved by comparing all values against the first.

Reply:
Hmm. I think doing "almost constant" correctly would be expensive and might be overkill. You'd have to compare every value with every other value, wouldn't you? Consider

`a ~= b && b ~= c`

does not imply

`a ~= c`

Reply:
I don't think you would have to do that. a_0 ~= x in [a_1..a_n] implies the field is constant.

Reply:
You are correct. Done.